TOOLS/lua/autocrop.lua

```diff
@@ -107,8 +128,21 @@ function do_crop()
         )
     end
     -- remove the cropdetect filter
-    del_filter_if_present(cropdetect_label)
+    remove_crop(cropdetect_label)
     timer=nil
 end
 
-mp.add_key_binding("C", "auto_crop", autocrop_start)
+function auto_corp()
+    local duration = mp.get_property_native("duration")
+    local percent = mp.get_property_native("percent-pos")
+    if options.auto == "yes"
+        and not vf_exist(crop_label)
```

Comment:
If "resume-playback" is enabled, applying the filter again is often useless or even an error.

I tried adding
`mp.observe_property("eof-reached", "bool", function(name, value)
    if value then
        remove_filter(crop_label)
    end
end)`
But doesn't work.
Is there any good solution?

Reply:
I don't think we should trust any previous crop result from resume-playback because the user may have changed parameters or the script may have otherwise changed. What do you mean by "error"?  If you see problems, we should iron those out.

You could try mp.register_event("end-file", cleanup) and have cleanup remove all the filters and delete the timers. I think think cleanup() should also be called at all the entry points: auto_crop, start_crop. For example if you hit "C" while the auto_delay timer is still running, you should delete that timer and just run detect_crop() right away. Similarly, if you flip to the next file in the playlist before all the timers have fired everything should start from scratch on the new file.

Reply:
How to cancel a timer in lua？
https://mpv.io/manual/master/#lua-scripting-mp-add-timeout(seconds,-fn)
I don’t see anything about it.

Reply:
```
timer = mp.add_timeout(4, timeout)
timer:kill()
```